utils.go

```diff
@@ -98,3 +98,31 @@ func WriteFile(fs Filesystem, filename string, data []byte, perm os.FileMode) er
 
 	return err
 }
+
+// CopyPath copies file across filesystems.
```

Comment:
I think we decided not to provide a `CopyFile` (or `CopyPath`) in the public API, since it is tricky to provide consistent behavior across implementations ([see this](https://github.com/golang/go/issues/8868)). So we'll limit ourselves to functions that exist in the standard library (e.g. `os`) for which we can provide sane behavior. So this would be private.